library/hls/src/main/java/com/google/android/exoplayer2/source/hls/HlsMediaChunk.java

```diff
@@ -291,13 +299,39 @@ public boolean isLoadCompleted() {
     return loadCompleted;
   }
 
+  /**
+   * Return the indexes of the first samples from each sample queue for this media chunk
+   *
+   * @return sample index {@link SampleQueue#getWriteIndex()}
+   */
+  int [] getFirstSampleIndexes() {
+    return firstSampleIndexes;
+  }
+
+  /**
+   * Set the index of the first sample written to a sample queue from this media chunk,
+   * indexed by the caller's stream index for the sample queue
+   *
+   * @param streamIndex - caller's index for the {@link SampleQueue}
+   * @param firstSampleIndex - index value to store
+   */
+  void setFirstSampleIndex(int streamIndex, int firstSampleIndex) {
```

Comment:
Yes, the non-chunkless prepare state causes this method to be called from the loader thread (the Extractor).   The `sampleQueue` array itself is accessed from both the player thread and loader threads because of non-chunkless prepare. 

The `ConcurrentModificationException` was because of the way `Extractor` init worked in pre-2.11.x versions, it looks like this is no longer so in the current `dev-v2`